pkg/probermanager/manager.go

```diff
@@ -0,0 +1,188 @@
+package probermanager
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/exaring/matroschka-prober/pkg/config"
+	"github.com/exaring/matroschka-prober/pkg/prober"
+	"github.com/prometheus/client_golang/prometheus"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type proberIndex struct {
+	path string
+	tos  uint8
+}
+
+type ProberManager struct {
+	probers   map[proberIndex]*prober.Prober
+	probersMu sync.RWMutex
+}
+
+func New() *ProberManager {
+	return &ProberManager{
+		probers: make(map[proberIndex]*prober.Prober),
+	}
+}
+
+func (pm *ProberManager) Configure(cfg *config.Config) error {
+	pm.probersMu.Lock()
+	defer pm.probersMu.Unlock()
+
+	desiredProbers, err := getDesiredProbers(cfg)
+	if err != nil {
+		return fmt.Errorf("unable to create probers: %v", err)
+	}
+
+	pm.adjustProbers(desiredProbers)
+	return nil
+}
```

Comment:
if you want to reduce the time the mutex is exclusively locked, consider moving it around the part that actually needs the locking. afaict `getDesiredProbers` doesn't need any locking

alternatively you can also move the locking into `adjustProbers`

```suggestion
func (pm *ProberManager) Configure(cfg *config.Config) error {
	desiredProbers, err := getDesiredProbers(cfg)
	if err != nil {
		return fmt.Errorf("unable to create probers: %v", err)
	}

	pm.probersMu.Lock()
	pm.adjustProbers(desiredProbers)
	pm.probersMu.Unlock()

	return nil
}
```